File: server/models/sensordata.js
const mongoose = require("mongoose");

const sensorDataSchema = new mongoose.Schema({
  deviceId: String,
  averageTemperature: Number,
  averageHumidity: Number,
  averageSmokeLevel: Number,
  flameDetected: Boolean,
  date: Date,
});

module.exports = mongoose.model("SensorData", sensorDataSchema, "sensordatas");

File: app/iotapp/lib/models/sensor_data.dart
class SensorData {
  final double averageTemperature;
  final double averageHumidity;
  final int averageSmokeLevel;
  final bool flameDetected;
  final DateTime date;

  SensorData({
    required this.averageTemperature,
    required this.averageHumidity,
    required this.averageSmokeLevel,
    required this.flameDetected,
    required this.date,
  });

  factory SensorData.fromJson(Map<String, dynamic> json) {
    return SensorData(
      averageTemperature: (json['averageTemperature'] ?? 0).toDouble(),
      averageHumidity: (json['averageHumidity'] ?? 0).toDouble(),
      averageSmokeLevel: (json['averageSmokeLevel'] ?? 0).toInt(),
      flameDetected: json['flameDetected'] ?? false,
      date: json['date'] != null ? DateTime.parse(json['date']) : DateTime.now(),
    );
  }
}
